m3/src/main/java/com/uber/m3/tally/m3/M3Reporter.java

```diff
@@ -524,6 +519,13 @@ public Thread newThread(Runnable r) {
 
         @Override
         public void run() {
+            try {
+                processorsShutdownSema.acquire();
```

Comment:
This actually has the potentially beneficial side effect of enforcing mutual exclusion between new and old processor, which I think is good.